activitychart-basic.html

```diff
@@ -0,0 +1,274 @@
+            }
+          }
+        }
+      },
+
+      attached: function() {
+        if(this.dataurl != "") {
+          this.$.ajaxCaller.generateRequest();
+        }
+        else {
+          this._plotDiagram();
+        }
+      },
+      /**
+      * Transforms the transmitted data into a plot
+      *
+      * @param {Object} data The data to be plotted
+      */
+      handleResponse: function (data) {
+        var goodData = this._normalizeJSON(data.detail.response);
+        this.dataset = goodData[0]["y"];
+        this.labels = goodData[0]["x"];
+        this._plotDiagram();
+      },
+
+      _buildDataset: function(labels, dataset) {
+        // create data like this:
+        // [ {'x': timeStamp0, 'y': value0}, ... ,{'x': timeStampN, 'y': valueN}]
+        var data = [];
+        for(var j = 0; j < labels.length; j++) {
+          data[j] = {'x': labels[j], 'y': dataset[j]};
+        }
+        return data;
+      },
+
+      _plotDiagram: function () {
+        console.log("Plotting...");
+        console.log(this.dataset);
+        console.log(this.labels);
+        var diagramDiv = this.$.diagram;
+
+        var data = this._buildDataset(this.labels, this.dataset);
+
+        var dataset = this.dataset;
+
+        var datasetMin;
+        if(this.minValue == null) {
+          datasetMin = Math.min(...dataset);
+        } else {
+          datasetMin = this.minValue;
+        }
+
+        var datasetMax;
+        if(this.maxValue == null) {
+          datasetMax = Math.max(...dataset);
+        } else {
+          datasetMax = this.maxValue;
+        }
+
+        var svgWidth = Math.ceil(dataset.length / this.squaresPerColumn) * this.squareSpacing,
+        svgHeight = this.squaresPerColumn * this.squareSpacing;
+
+        //so far, the appropriate color is calculated based on the value of the datum, there is no set of fixed colors
+        var colorScale = d3.scaleLinear()
+          .domain([datasetMin, datasetMax])
+          .range([this.accentcolor, this.primarycolor]);
+
+        var svg = d3.select('#diagram').append('svg')
+          .attr('width', svgWidth)
+          .attr('height', svgHeight);
+
+        var squares = svg.selectAll('rect')
+          .data(data)
+          .enter().append('rect')
+          .attr('width', this.squareWidth)
+          .attr('height', this.squareWidth)
+          .attr('x', this._spaceSquareHor)
+          .attr('y', this._spaceSquareVert)
+          .attr('fill', function(d) { return colorScale(d.y); })
+          .on('mouseover', this._showLabel)
+          .on('mouseout', this._hideLabel);
+      },
+
+      _showLabel: function (d, i) {
+        //using a group element 'label' with an appended text element and a rect element as background to display a label
+
+        var polyElement = Polymer.dom(this).parentNode.parentNode.parentNode;
```

Comment:
is there maybe a better way to traverse the dom than `.parentNode.parentNode.parentNode`? Seems kinda odd to me, definitely don't know better though... :sweat_smile: